config/validate.go

```diff
@@ -75,99 +73,145 @@ func ValidateConfigFile(file string) ([]UnexpectedKey, error) {
 	var c Config
 	candidates := makeCandidates(reflect.TypeOf(c))
 
-	var parentKeys []string
+	var parentConfKeys []string
 	configFields := reflect.VisibleFields(reflect.TypeOf(c))
 	for _, f := range configFields {
 		if s := f.Tag.Get("conf"); s == "parent" {
-			parentKeys = append(parentKeys, normalizeKeyname(f))
+			parentConfKeys = append(parentConfKeys, normalizeKey(f))
 		}
 	}
 
 	var unexpectedKeys []UnexpectedKey
-	for _, v := range md.Undecoded() {
+	var detectedKeys []string
+
+	undecodedKeys := md.Undecoded()
+	sort.Slice(undecodedKeys, func(i, j int) bool {
+		return undecodedKeys[i].String() < undecodedKeys[j].String()
+	})
+
+	for _, v := range undecodedKeys {
+		/*
+			```
+			[plugin.checks.incorrect]
+			command = "test command"
+			action = { command = "test command", user = "test user", en = { TEST_KEY = "VALUE_1" }
+
+			[plugins.check.incorrect]
+			command = "test command"
+			```
+
+			undecodedKeys -> [
+				plugin.checks.incorrect.action.en,
+				plugin.checks.incorrect.action.en.TEST_KEY,
+				plugins.check.incorrect,
+				plugins.check.incorrect.command,
+			]
+
+			v: plugin.checks.incorrect.action.en
+			splitedKey -> [plugin, checks, incorrect, action, en]
+			topKey -> plugin
+			lastKey -> en
+			parentKey -> plugin.checks.incorrect.action
+
+			v: plugin.checks.incorrect.action.en.TEST_KEY
+			splitedKey -> [plugin, checks, incorrect, action, en, TEST_KEY]
+			topKey -> plugin
+			lastKey -> TEST_KEY
+			parentKey -> plugin.checks.incorrect.action.en
+
+			v: plugins.check.incorrect
+			splitedKey -> [plugins, check, incorrect]
+			topKey -> plugins
+			lastKey -> incorrect
+			parentKey -> plugins.check
+
+			v: plugins.check.incorrect.command
+			splitedKey -> [plugins, check, incorrect, command]
+			topKey -> plugins
+			lastKey -> command
+			parentKey -> plugins.check.incorrect
+		*/
```

Comment:
I think only a simple example of each would suffice to explain. Were the procedure so complex that all test cases must be exemplified, it would have to be divided into functions with unit tests.